server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java

```diff
@@ -431,23 +435,29 @@ private CommandExecutor startCommandExecutor(
             }
         };
 
+        statusManager = new ServerStatusManager(cfg.dataDir());
         final CommandExecutor executor;
         final ReplicationMethod replicationMethod = cfg.replicationConfig().method();
         switch (replicationMethod) {
             case ZOOKEEPER:
-                executor = newZooKeeperCommandExecutor(pm, repositoryWorker, meterRegistry, sessionManager,
-                                                       onTakeLeadership, onReleaseLeadership);
+                executor = newZooKeeperCommandExecutor(pm, repositoryWorker, statusManager, meterRegistry,
+                                                       sessionManager, onTakeLeadership, onReleaseLeadership);
                 break;
             case NONE:
                 logger.info("No replication mechanism specified; entering standalone");
-                executor = new StandaloneCommandExecutor(pm, repositoryWorker, sessionManager,
+                executor = new StandaloneCommandExecutor(pm, repositoryWorker, statusManager, sessionManager,
                                                          cfg.writeQuotaPerRepository(),
                                                          onTakeLeadership, onReleaseLeadership);
                 break;
             default:
                 throw new Error("unknown replication method: " + replicationMethod);
         }
 
+        final ServerStatus initialServerStatus = statusManager.serverStatus();
+        executor.setWritable(initialServerStatus.writable());
```

Comment:
`ProjectInitializer.initializeInternalProject()` is added after this method returns.
https://github.com/line/centraldogma/blob/48586150dbf1370e3739b7fda7ea00cef47aa092/server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java#L378-L381
If `writable` is `false`, the internal repositories are not created. It is an undesirable status. 

However, this is an administrative operation to make a repository read-only that new servers do not need because they are empty. So it would be safe that all new servers must have `writable == true`.